### PR TITLE
Add a `fuel.nix` mdbook

### LINKED ISSUE #59: Update README to recommend the Determinate Systems nix-installer
## Motivation

The [Determinate Systems nix-installer](https://github.com/DeterminateSystems/nix-installer) tool provides an opinionated and much simpler installation process that nicely aligns with the way fuel.nix encourages using Nix.

Namely, it enables the `flakes` and `nix-command` features by default, provides a much more concise installation plan description, and also offers non-interactive installation options. The tool is built in Rust, and also plans to provide the tool as a library, which also makes Nix a strong candidate as a basis for fuelup.

## Implementation

In a single command, users can install Nix *and* enable our binary cache with something like the following:

```console
curl --proto '=https' --tlsv1.2 -sSf -L https://install.determinate.systems/nix | sh -s -- install --extra-conf "extra-substituters = https://mitchmindtree-fuellabs.cachix.org" --extra-conf "extra-trusted-public-keys = mitchmindtree-fuellabs.cachix.org-1:UDUQvwjM3wRCZe1chrgqAehb3M0M5x9qjpEwJwPn7Ik="
```

I've been unable to confirm this works yet as the `--extra-conf` flags are currently not working (on nix-installer `v0.8.0`) due to https://github.com/DeterminateSystems/nix-installer/issues/427, however this has since been fixed in https://github.com/DeterminateSystems/nix-installer/pull/456 and should become available as a part of the [upcoming `v0.8.1` release](https://github.com/DeterminateSystems/nix-installer/milestone/11).

Once this version is released and the installation process can be confirmed to succeed on a fresh Ubuntu VM along with fetching the latest fuel binaries from successfully from cache, we should update the README with this recommended approach.

### PR DESCRIPTION
**Pull request by mitchmindtree**
This adds a new `book/` directory containing an mdbook for `fuel.nix`.

The new book will replace the README with a much friendlier "Quick Start", and a more searchable and detailed guide covering how to install Nix, use fuel packages, and how to contribute to fuel.nix itself.

Closes #59.

TODO
----

- [x] Add a new "Quick Start" section with the new two-step installation approach.
- [x] Merge existing README content into subchapters.
- [x] Add a more detailed "Nix Setup" chapter for those interested in more details.
- [x] Add a section on `nix profile install` for persistent installation to the "Packages" chapter.
- [x] Add a new "Contributing" section with details on how to add/update packages and milestones.
- [x] Update the README to simply point to the book and Quick Start.
- [x] Add a CI step to publish the book.
- ~Open an issue to remove the **NOTE:** under the `nix profile install` section required due to DeterminateSystems/nix-installer#477.~ It seems this issue only occurs if a previous Nix installation existed but was not cleaned up properly. The `note` has been updated.

### CODE REVIEW

**Comment by mitchmindtree** [comment]
This is just about ready to go, however it seems like the recent [addition of the `sysinfo` crate in Sway and forc for benchmarking](https://github.com/FuelLabs/sway/pull/4564) may have introduced some non-determinism in the macOS builds.

[Today's `refresh-manifests` job succeeded](https://github.com/FuelLabs/fuel.nix/actions/runs/5116169386), however the current CI `macos-latest` matrix job is missing the cache, and [failing with a linker error](https://github.com/FuelLabs/fuel.nix/actions/runs/5119801131/jobs/9205518400#step:5:3347).

```console
       >   = note: Undefined symbols for architecture x86_64:
       >             "_kCFURLVolumeAvailableCapacityForImportantUsageKey", referenced from:
       >                 _$LT$sysinfo..apple..disk..Disk$u20$as$u20$sysinfo..traits..DiskExt$GT$::refresh::ha090b62f02761b2b in libsysinfo-bfbd6883445a3969.rlib(sysinfo-bfbd6883445a3969.sysinfo.4f5fc5ae-cgu.15.rcgu.o)
       >                 sysinfo::apple::disk::get_disks::h63aaac400a8fb64f in libsysinfo-bfbd6883445a3969.rlib(sysinfo-bfbd6883445a3969.sysinfo.4f5fc5ae-cgu.15.rcgu.o)
       >           ld: symbol(s) not found for architecture x86_64
       >           clang-11: error: linker command failed with exit code 1 (use -v to see invocation)
       >           
       >
       > error: could not compile `forc` due to previous error
       For full logs, run 'nix log /nix/store/vgq1zhv6q43wbgbd5hifm26ab2wikwj9-forc-0.39.1.drv'.
error: 1 dependencies of derivation '/nix/store/jfxshavhlsxkkdfnjdqpxfcyjv1icj6f-fuel-nightly.drv' failed to build
```

All that said, it's unclear how the latest `refresh-manifests` action succeeded without incurring the same linker error, or if `sysinfo` is definitely the cause of the cache invalidation :thinking:

**Comment by mitchmindtree** [comment]
It looks like the same issue was encountered in `nushell` https://github.com/NixOS/nixpkgs/issues/214872. Apparently fixed in https://github.com/NixOS/nixpkgs/pull/215782. It could be worth trying a similar patch.

**Comment by mitchmindtree** [comment]
The `fuel-nightly` `macos-latest` failure appears unrelated, tracking under https://github.com/FuelLabs/fuel.nix/issues/64, will investigate shortly!

File: .github/workflows/publish-book.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+      - uses: actions/checkout@v3
+      - name: build
+        run: nix develop --print-build-logs --no-update-lock-file .#book-dev --command mdbook build book
+      - name: deploy
+        uses: peaceiris/actions-gh-pages@v3
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          publish_dir: ./book/book
+          destination_dir: ./
+          cname: nix.fuel.network
```
Comment (mitchmindtree) [inline .github/workflows/publish-book.yml:21]: @Voxelot do you know what we might need to do with the `fuel.network` hosting to get this `cname` to work like we have for [rust.fuel.network](https://github.com/FuelLabs/fuels-rs/blob/002bc8f123e749d77e6681ff35ed4fba480ede59/.github/workflows/gh-pages.yml#L29)?

**Review by mitchmindtree — COMMENTED** [review]
OK this should be ready to go!

I'll open a follow-up issue/PR for adding markdown linting, but want to try to address #70 first.

**Comment by mitchmindtree** [comment]
I *think* this should at least publish to the gh-pages branch and the github.io link for now - I'll open up a follow-up issue for the `cname` redirect.

**Review by JoshuaBatty — APPROVED** [review]

**Pull request merged**